test/e2e/framework/volume_util.go

```diff
@@ -397,13 +397,43 @@ func VolumeTestCleanup(f *Framework, config VolumeTestConfig) {
 	}
 }
 
-// Start a client pod using given VolumeSource (exported by startVolumeServer())
+// TestVolumeClient start a client pod using given VolumeSource (exported by startVolumeServer())
 // and check that the pod sees expected data, e.g. from the server pod.
 // Multiple VolumeTests can be specified to mount multiple volumes to a single
 // pod.
 func TestVolumeClient(client clientset.Interface, config VolumeTestConfig, fsGroup *int64, fsType string, tests []VolumeTest) {
-	By(fmt.Sprint("starting ", config.Prefix, " client"))
+	By(fmt.Sprint("starting ", config.Prefix, "-client"))
 	var gracePeriod int64 = 1
+	var commands []string
+	var fileCheckCommand string
+	var securityContext *v1.PodSecurityContext
+	var nodeSelector map[string]string
+	var image string
+	if !NodeOSDistroIs("windows") {
+		commands = []string{
+			"/bin/sh",
+			"-c",
+			"while true ; do cat /opt/0/index.html ; sleep 2 ; ls -altrh /opt/  ; sleep 2 ; done ",
+		}
+		securityContext = &v1.PodSecurityContext{
+			SELinuxOptions: &v1.SELinuxOptions{
+				Level: "s0:c0,c1",
+			},
+			FSGroup: fsGroup,
+		}
+		fileCheckCommand = "cat"
+		nodeSelector = config.NodeSelector
+		image = BusyBoxImage
+	} else {
+		commands = []string{
+			"powershell",
+			"/c",
+			"while(1) {cat /opt/0/index.html ; sleep 2 ; ls /opt/; sleep 2}",
+		}
+		fileCheckCommand = "powershell /c type"
+		nodeSelector = map[string]string{"beta.kubernetes.io/os": "windows"}
```

Comment:
Can this be set in gce pd driver config instead of in the test case?

Reply:
fixed

Reply:
fixed